File: aqi-estimator.client/src/app/shared/classes/models/outgoing/AQIPredictionRequest.ts
export class AQIPredictionRequest {
    indata: string;
    lat: number | undefined | null;
    lon: number | undefined | null;

    constructor(_indata: string, _lat: number | undefined | null, _lon: number | undefined | null) {
        this.indata = _indata;
        this.lat = _lat;
        this.lon = _lon;
    }
}

File: aqi-estimator.server/src/main/java/com/codefu/refactor/api/aqiestimatorserver/DTO/clientRequest/AQIPredictionRequest.java
package com.codefu.refactor.api.aqiestimatorserver.DTO.clientRequest;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.NonNull;

import java.util.Base64;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class AQIPredictionRequest {
    private String indata;
    private Double lat;
    private Double lon;

    public byte[] getIndataAsBytes() {
        return Base64.getDecoder().decode(indata);
    }
}
